pages/article/ArticleHandler.inc.php

```diff
@@ -342,7 +345,11 @@ public function view($args, $request)
             }
 
             if (!HookRegistry::call('ArticleHandler::view', [&$request, &$issue, &$article, $publication])) {
-                return $templateMgr->display('frontend/pages/article.tpl');
+                $templateMgr->display('frontend/pages/article.tpl');
+                if ($article->getData('status') == Submission::STATUS_PUBLISHED && !$request->isDNTSet()) {
+                    event(new UsageEvent(Application::ASSOC_TYPE_SUBMISSION, $article->getId(), $context->getId(), $article->getId(), null, null, $publication->getData('issueId')));
```

Comment:
I've just changed it so, but I am not sure if we should fire an event if the object is not published -- this will fire an event also when an editor previews an issue or submission... and I am not sure why would this be good... :thinking: 
@asmecher, what do you think?

Reply:
Yep! I'd prefer to decouple the event from the logic we use when processing that event. The model should be: one event, many listeners.